src/Hamiltonians/Molecular.jl

```diff
@@ -0,0 +1,613 @@
+"""
+    MolecularMolecularHamiltonian(fcidump::String[, starting_address::FermiFS2C])
+    
+    MolecularMolecularHamiltonian(fd::ElemCo.FciDumps.FDump[, starting_address::FermiFS2C])
+
+Implements a Molecular Hamiltonian with [ElemCo.jl](https://elem.co.il). 
+
+The expression of Hamiltonian under Bohn-Oppenheimer approximation is 
+```math
+\\begin{aligned}
+    \\hat{H}   & = \\hat{H}_0 + \\hat{H}_1 + \\hat{H}_2, \\\\
+    \\hat{H}_1 & = \\sum_{i,j} h_{ij} a^{\\dagger}_j a_i, \\\\
+    \\hat{H}_2 & = \\sum_{kl,ij} V_{kl,ij} a^{\\dagger}_{k} a^\\dagger_{l} a_j a_i.
+\\end{aligned}
+```
+
+# Arguments
+* `fcidump`: The path to FCIDUMP file of molecular.
+* `fd`: The FCIDUMP structure defined in [ElemCo.jl](https://elem.co.il).
+* `starting_address`: The starting address, defines number of particles and sites.
+
+See also [`FermiFS2C`](@ref)
+
+!!! warning
+    The FCIDUMP file is assumed to be a Restricted Hartree-Fock(RHF) Hamiltonian.
+"""
+struct MolecularHamiltonian{T,A<:FermiFS2C,D<:FDump} <: AbstractHamiltonian{T}
+    starting_address::A
+    fcidump::D
+end
```

Comment:
It would be nice to have a method for `show(io, h::MolecularHamiltonian)` that is really concise and doesn't print out all of the fcidump to make working on the REPL a bit nicer. Maybe printing the `starting_address` is sufficient?

It could also be nice if the type instance remembered which molecule this was for, maybe with a `specifier::String` field that could either hold the path to the FCIDUMP file or be a user-supplied name. Maybe in the future this could be uplifted from information in the `AtomsBase.jl` package.